File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/command/CategoryCommand.java
package dev.jefy.connectpro.management.appliacaion.command;

import org.jspecify.annotations.NullMarked;

import dev.jefy.connectpro.management.appliacaion.dtos.CategoryRequest;
import dev.jefy.connectpro.management.domain.vo.CategoryId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface CategoryCommand {
    CategoryId create(CategoryRequest request);
    CategoryId update(CategoryId categoryId, CategoryRequest request);
    void delete(CategoryId categoryId);
}

File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/command/CategoryCommand.kt
package dev.jefy.connectpro.management.appliacaion.command

import dev.jefy.connectpro.management.appliacaion.dtos.CategoryRequest
import dev.jefy.connectpro.management.domain.vo.CategoryId
import org.jspecify.annotations.NullMarked

/**
 * @author Jôph Yamba
 */

interface CategoryCommand {
    fun create(request: CategoryRequest): CategoryId
    fun update(categoryId: CategoryId, request: CategoryRequest): CategoryId
    fun delete(categoryId: CategoryId)
}
